src/DssSpell.t.sol

```diff
@@ -875,4 +888,261 @@ contract DssSpellTest is DssSpellTestBase {
+
+            uint256 pbalance = sky.balanceOf(address(this));
+            skip(7 days);
+            rewards.getReward();
+            assertGt(sky.balanceOf(address(this)), pbalance);
+
+            vm.revertTo(before);
+        }
+
+        // Check if distribute can be called again in the future
+        {
+            uint256 before = vm.snapshot();
+
+            uint256 pbalance = sky.balanceOf(address(rewards));
+            skip(7 days);
+            dist.distribute();
+            assertGt(sky.balanceOf(address(rewards)), pbalance, "testUsdsSkyRewards/distribute-no-increase-balance");
+
+            vm.revertTo(before);
+        }
+    }
+
+    function testUsds01Rewards() public {
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done(), "TestError/spell-not-done");
+
+        StakingRewardsLike rewards = StakingRewardsLike(addr.addr("REWARDS_USDS_01"));
+
+        // Sanity checks
+        assertEq(rewards.rewardsDistribution(), address(0),    "testUsds01Rewards/rewards-invalid-rewards-dist");
+        assertEq(rewards.stakingToken(),        address(usds), "testUsds01Rewards/rewards-staking-token-mismatch");
+        assertEq(rewards.rewardsToken(),        address(0),    "testUsds01Rewards/rewards-invalid-rewards-token");
+
+
+        // Check if users can stake
+        {
+            uint256 before = vm.snapshot();
+
+            uint256 stakingWad = 100_000 * WAD;
+            _giveTokens(address(usds), stakingWad);
+            usds.approve(address(rewards), stakingWad);
+            rewards.stake(stakingWad);
+            assertEq(rewards.balanceOf(address(this)), stakingWad, "testUsds01Rewards/rewards-invalid-staked-balance");
+
+            vm.revertTo(before);
+        }
+    }
+
+    function testVestedRewardsDistributionJob() public {
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done(), "TestError/spell-not-done");
+
+        SequencerLike seq = SequencerLike(addr.addr("CRON_SEQUENCER"));
+        VestedRewardsDistributionJobLike job = VestedRewardsDistributionJobLike(addr.addr("CRON_REWARDS_DIST_JOB"));
+        VestedRewardsDistributionLike distUsdsSky = VestedRewardsDistributionLike(addr.addr("REWARDS_DIST_USDS_SKY"));
+
+        // Since distribute() was called in the spell, initially the job is not due for execution...
+        {
+            // `workable()` is allowed to modify state in this job, so we need to make sure to revert its effects.
+            uint256 before = vm.snapshot();
+
+            (bool ok, ) = job.workable(seq.getMaster());
+            assertFalse(ok, "testVestedRewardsDistributionJob/unexpected-due-job");
+
+            vm.revertTo(before);
+        }
+
+        // ... as we skip to the next epoch, then we can execute
+        skip(job.intervals(address(distUsdsSky)));
```

Comment:
This test (or any other one as far as I can see) doesn't yet check `intervals[dist]` value to be set to `7 days - 1 hours`. I would instead:

1. Check that the job is not executable when skipping to (`7 days - 1 hours - 1`)
2. Skip to the hardcoded `7 days - 1 hours`

Reply:
I'm actually not sure why skipping to `job.intervals(address(distUsdsSky))` allows claiming rewards, since it should currently return `7 days - 1 hours` which should be not enough?

Reply:
Updated.

Reply:
It is enough... distributions can be made at any point in time, the only enforcement is the job frequency, which is set to `7 days - 1 hours`. We use strict inequality in the job, meaning that if we skip the exact interval, we are allowed to call it.